src/Constants_Conversion.f90

```diff
@@ -232,4 +236,130 @@ ELEMENTAL FUNCTION K_to_F(in_temp) RESULT(out_temp)
+  ENDSELECT
+
+ENDFUNCTION tempTo_K
+!
+!-------------------------------------------------------------------------------
+!> @brief converts temperature to Kelvin from Kelvin, Rankine, or Fahrenheit,
+!> if any other unit is presented in_temp is simply returned
+!> @param in_temp the temperature input as the specified units
+!> @param units character indicating the units of in_temp
+!> @return out_temp the temperature converted to Celsius
+!>
+ELEMENTAL FUNCTION tempTo_C(in_temp,units) RESULT(out_temp)
+  REAL(SRK),INTENT(IN) :: in_temp
+  CHARACTER(LEN=1),INTENT(IN) :: units
+  REAL(SRK) :: out_temp
+
+  SELECTCASE(units)
+  CASE('K')
+    out_temp = in_temp - C2K
+  CASE('R')
+    out_temp = in_temp * R2K - C2K
+  CASE('F')
+    out_temp = (in_temp - 32.0_SRK) * R2K
+  CASE DEFAULT
+    out_temp = in_temp
+  ENDSELECT
+
+ENDFUNCTION tempTo_C
+!
+!-------------------------------------------------------------------------------
+!> @brief converts temperature to Kelvin from Celsius, Rankine, or Kelvin,
+!> if any other unit is presented in_temp is simply returned
+!> @param in_temp the temperature input as the specified units
+!> @param units character indicating the units of in_temp
+!> @return out_temp the temperature converted to Fahrenheit
+!>
+ELEMENTAL FUNCTION tempTo_F(in_temp,units) RESULT(out_temp)
+  REAL(SRK),INTENT(IN) :: in_temp
+  CHARACTER(LEN=1),INTENT(IN) :: units
+  REAL(SRK) :: out_temp
+
+  SELECTCASE(units)
+  CASE('K')
+    out_temp = (in_temp - C2K) / R2K + 32.0_SRK
+  CASE('R')
+    out_temp = in_temp - F2R
+  CASE('C')
+    out_temp = (in_temp / R2K) + 32.0_SRK
+  CASE DEFAULT
+    out_temp = in_temp
+  ENDSELECT
+
+ENDFUNCTION tempTo_F
+!
+!-------------------------------------------------------------------------------
+!> @brief converts temperature to Kelvin from Celsius, Rankine, or Fahrenheit,
+!> if any other unit is presented in_temp is simply returned
+!> @param in_temp the temperature input as the specified units
+!> @param units character indicating the units of in_temp
+!> @return out_temp the temperature converted to Fahrenheit
+!>
+ELEMENTAL FUNCTION tempTo_R(in_temp,units) RESULT(out_temp)
+  REAL(SRK),INTENT(IN) :: in_temp
+  CHARACTER(LEN=1),INTENT(IN) :: units
+  REAL(SRK) :: out_temp
+
+  SELECTCASE(units)
+  CASE('K')
+    out_temp = in_temp / R2K
+  CASE('F')
+    out_temp = in_temp + F2R
+  CASE('C')
+    out_temp = (in_temp + C2K) / R2K
+  CASE DEFAULT
+    out_temp = in_temp
+  ENDSELECT
+
+ENDFUNCTION tempTo_R
+!
+!-------------------------------------------------------------------------------
+!> @brief converts temperature to Kelvin from Celsius, Rankine, or Fahrenheit,
+!> if any other unit is presented in_temp is simply returned
+!> @param in_temp the temperature input as the specified units
+!> @param units character indicating the units of in_temp
+!> @return out_temp the temperature converted to Fahrenheit
+!>
+ELEMENTAL FUNCTION convertTemp(in_temp,in_units,out_units) RESULT(out_temp)
```

Comment:
Good call on this one, but need to make it public

Reply:
yeah, this is a good one